### PR TITLE
Cradio

### PR DESCRIPTION
**Pull request by davidphilipbarr**
34 key keyboard with no diodes.

### CODE REVIEW

File: app/boards/shields/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+config ZMK_KEYBOARD_NAME
+	default "cradios"
+
+# Unable to use interrupts as the same pin number is used
+# across A & B controllers, and STM32F303CCT6 can't enable
+# interrutps for multiple controllers for the same "line"
+# for the external interrupts.
+config ZMK_KSCAN_GPIO_POLLING
```
Comment (petejohanson) [inline app/boards/shields/Kconfig.defconfig:11]: Were you testing with proton-c controllers? If so, there's actually a way to provide board specific overrides for a given shield, you can see that in https://docs.zephyrproject.org/2.3.0/guides/porting/shields.html?highlight=shield#board-specific-shield-configuration

I would remove this from the "base" defconfig here, and put it in the proton-c specific one.

File: app/boards/shields/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (petejohanson) [inline app/boards/shields/Kconfig.defconfig:1]: Please add the standard ZMK header here, e.g. 

```
# Copyright (c) 2020 The ZMK Contributors
# SPDX-License-Identifier: MIT
```

File: app/boards/shields/Kconfig.shield [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2020 Pete Johanson
```
Comment (petejohanson) [inline app/boards/shields/Kconfig.shield:1]: We're still working on fixing up existing boards, but please use the "single author file" version like so:

```suggestion
# Copyright (c) 2020 The ZMK Contributors
```

File: .gitignore [hunk 4]
```diff
@@ -4,3 +4,4 @@
 /tools
 /zephyr
 /build
+**/.DS_Store
```
Comment (petejohanson) [inline .gitignore:7]: Thanks!

File: app/boards/shields/cradio/Kconfig.shield [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2020 Pete Johanson
```
Comment (petejohanson) [inline app/boards/shields/cradio/Kconfig.shield:1]: Ditto.

```suggestion
# Copyright (c) 2020 The ZMK Contributors
```

File: app/boards/shields/cradio/cradio.dtsi [hunk 6]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
```
Comment (petejohanson) [inline app/boards/shields/cradio/cradio.dtsi:2]: Same:

```suggestion
 * Copyright (c) 2020 The ZMK Contributors
```

File: app/boards/shields/cradio/cradio.dtsi [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+	
+	kscan0: kscan {
+		compatible = "zmk,kscan-gpio-direct";
+		label = "KSCAN";
+	};
+	
+	
+	
+	
+	bt_unpair_combo: bt_unpair_combo {
```
Comment (petejohanson) [inline app/boards/shields/cradio/cradio.dtsi:35]: We're working on removing this feature, so you can skip this whole node for new shields.

File: app/boards/shields/cradio/cradio.dtsi [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+		compatible = "zmk,bt-unpair-combo";
+	};
+	
+		};
+		
+		&kscan0 {
```
Comment (petejohanson) [inline app/boards/shields/cradio/cradio.dtsi:41]: This should be merged with the initial `kscan0` definition above this.

File: app/boards/shields/cradio/cradio_left.overlay [hunk 9]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
+ *
+ * SPDX-License-Identifier: MIT
+ */
+#include "cradio.dtsi"
+
+&bt_unpair_combo {
```
Comment (petejohanson) [inline app/boards/shields/cradio/cradio_left.overlay:8]: Can be removed.

File: app/boards/shields/cradio/cradio.keymap [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+#include <dt-bindings/zmk/keys.h>
+
+
+/ {
+        keymap {
+                compatible = "zmk,keymap";
+
+                default_layer {
+		bindings = <
+		&kp Q &kp W &kp E &kp R &kp T 		&kp Y &kp U  &kp I    &kp O   &kp P
```
Comment (petejohanson) [inline app/boards/shields/cradio/cradio.keymap:11]: Seems like a "too simple" stock keymap. Obviously, a board like this is likely to be customized pretty specifically by each user, so not a huge deal if we just leave it as is.

File: app/boards/shields/cradio/cradio_right.overlay [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * SPDX-License-Identifier: MIT
+ */
+#include "cradio.dtsi"
+
+&default_transform {
+	col-offset = <17>;
+};
+
+&bt_unpair_combo {
```
Comment (petejohanson) [inline app/boards/shields/cradio/cradio_right.overlay:12]: Ditto.

File: app/boards/shields/cradio/default.keymap [hunk 12]
```diff
@@ -0,0 +1,19 @@
+#include <behaviors.dtsi>
```
Comment (petejohanson) [inline app/boards/shields/cradio/default.keymap:1]: This is an older file naming, and isn't used any more. The previous `cradio.keymap` you added is the one that will actually be used by ZMK. Remove this one please.

**Review by petejohanson — CHANGES_REQUESTED** [review]
Thanks for the PR. A mixture of items from header tweaks to some shield details.

File: app/boards/shields/cradio/cradio_right.overlay [hunk 13]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
```
Comment (petejohanson) [inline app/boards/shields/cradio/cradio_right.overlay:2]: Duplicate here too.

**Review by petejohanson — COMMENTED** [review]
A few follow ups, in addition to the few outstanding items. Thanks!

File: app/boards/shields/Kconfig.defconfig (hunk 1 above)
Comment (davidphilipbarr) [reply app/boards/shields/Kconfig.defconfig:11]: I'm not clear what you mean here so i just deleted everything about config...

File: app/boards/shields/Kconfig.defconfig (hunk 2 above)
Comment (davidphilipbarr) [reply app/boards/shields/Kconfig.defconfig:1]: added

File: app/boards/shields/Kconfig.shield (hunk 3 above)
Comment (davidphilipbarr) [reply app/boards/shields/Kconfig.shield:1]: added
# Copyright (c) 2020 The ZMK Contributors
# SPDX-License-Identifier: MIT

File: app/boards/shields/cradio/Kconfig.shield (hunk 5 above)
Comment (davidphilipbarr) [reply app/boards/shields/cradio/Kconfig.shield]: added# Copyright (c) 2020 The ZMK Contributors
# SPDX-License-Identifier: MIT

File: app/boards/shields/cradio/cradio.dtsi (hunk 8 above)
Comment (davidphilipbarr) [reply app/boards/shields/cradio/cradio.dtsi:38]: i've nudged this around and it seem to still compile...

File: app/boards/shields/cradio/cradio.keymap (hunk 10 above)
Comment (davidphilipbarr) [reply app/boards/shields/cradio/cradio.keymap:18]: i would like to add a better example keymap, but it's going to rely heavily on mod taps and maybe chords, i'd like to modify it as these become more established and stabilised. (i'm using okke's branch)

File: app/boards/shields/cradio/Kconfig.defconfig [hunk 14]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2020 The ZMK Contributors
+# SPDX-License-Identifier: MIT
+
+config ZMK_KEYBOARD_NAME
+	default "cradio"
+
+config ZMK_KSCAN_GPIO_POLLING
```
[the remark this answers is not in the document]
Comment (davidphilipbarr) [reply app/boards/shields/cradio/Kconfig.defconfig]: gotcha, sorted, can't wait to try it.

File: app/boards/shields/cradio/cradio_left.overlay [hunk 15]
```diff
@@ -0,0 +1,6 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
```
[the remark this answers is not in the document]
Comment (davidphilipbarr) [reply app/boards/shields/cradio/cradio_left.overlay:2]: modded to mirror corne overlay header

File: app/boards/shields/cradio/cradio_right.overlay (hunk 13 above)
Comment (davidphilipbarr) [reply app/boards/shields/cradio/cradio_right.overlay:2]: modded to mirror corne overlay header

File: app/boards/shields/cradio/cradio.keymap (hunk 10 above)
Comment (petejohanson) [reply app/boards/shields/cradio/cradio.keymap:18]: Ok, that's fine, and makes sense.

Beyond that, my only concern is the lack of the `&bt` behavior bindings anywhere, so users can reset the bond, use different profiles, etc.

Thoughts on at least adding a second layer that includes those OOTB?

File: app/boards/shields/cradio/cradio_left.overlay (hunk 15 above)
[the remark this answers is not in the document]
Comment (petejohanson) [reply app/boards/shields/cradio/cradio_left.overlay:2]:
```suggestion
 * Copyright (c) 2020 The ZMK Contributors
```

The Corne overlay header is wrong, and needs fixing, which has been done in another PR that's nearly in.

File: app/boards/shields/cradio/cradio_right.overlay (hunk 13 above)
Comment (petejohanson) [reply app/boards/shields/cradio/cradio_right.overlay:2]:
```suggestion
 * Copyright (c) 2020 The ZMK Contributors
```

Ditto.

**Review by petejohanson — COMMENTED** [review]
Thanks for the revisions. I few lingering items and then we can get this merged.

**Review by petejohanson — APPROVED** [review]
Thanks!

File: .gitignore (hunk 4 above)
Comment (petejohanson) [reply .gitignore:7]: This sadly conflicts w/ `main` now. Can you revert this, then we can merge?

Comment (petejohanson) [inline .gitignore:7]:
```suggestion
```

**Pull request merged**